### ISSUE TITLE
Broken production with Metro bundler (ReferenceError)

### ISSUE THREAD

**Issue by zoontek** [FormidableLabs/urql#736]
Hi 👋

On the latest version, a modification ([somewhere around here](https://github.com/FormidableLabs/urql/blob/urql%401.9.7/packages/core/src/client.ts#L102)) broke the metro bundler in production mode (and possibly other production apps? It's not tied to minification, but I didn't tried on other bundlers).

<img src="https://user-images.githubusercontent.com/1902323/79981632-49273d00-84a5-11ea-9b7a-8dc9186ca76b.png" width="50%" />

#### urql version: `1.9.7`

**Steps to reproduce**

1. Create a new React Native project
1. Add `urql@1.9.7`, wrap your app in a `Provider` with an urql client
1. Build your app for production
1. Starts your app. It will crash at client init.

**Expected behavior**

No `JS ReferenceError`.

**Actual behavior**

For some reason, the variable is not declared.

### Debugging

In order to spot the issue, I created a server to serve the unminified production bundle to my debug application. Create a file at project root:

```js
// server.js

const express = require("express");
const app = express();

app.use((req, res, next) => {
  console.log(req.url);
  res.setHeader("Content-Type", "application/javascript");
  next();
});

app.use("/", express.static("."));
app.use("/assets", express.static("."));
app.get("/status", (_req, res) => {
  res.send("packager-status:running");
});

app.listen(8081, () => {
  console.log("Serving directory as localhost:8081");
});
```

Build a bundle with:

```sh
yarn react-native bundle --entry-file index.js --bundle-output index.bundle --platform ios --dev false --minify false
```

Then run the `server.js` file to fake metro bundler, it will be easier to debug.

### Fix

The missing declaration is `f`:

![Screenshot 2020-04-22 at 13 57 36](https://user-images.githubusercontent.com/1902323/79982266-55f86080-84a6-11ea-9ed9-87c57a146e34.png)

If I add it manually to the bundle, it works:

![Screenshot 2020-04-22 at 13 57 58](https://user-images.githubusercontent.com/1902323/79982006-e4201700-84a5-11ea-8e22-a46e866c0e32.png)

**Comment by JoviDeCroock** [comment]
Hey would you mind doing `yarn list --pattern "urql"` and pasting the output here please

**Comment by zoontek** [comment]
This one breaks:

```
yarn list v1.22.4
├─ @graphql-codegen/typescript-urql@1.13.3
├─ @urql/core@1.11.4
└─ urql@1.9.7
```

This one works:

```
yarn list v1.22.4
├─ @graphql-codegen/typescript-urql@1.13.3
├─ @urql/core@1.10.8
└─ urql@1.9.6
```

**Comment by JoviDeCroock** [comment]
Hmm, that's an odd one. In both builds this seems defined

[ESM](https://www.runpkg.com/?@urql/core@1.10.4/dist/urql-core.mjs#516)
[CJS](https://www.runpkg.com/?@urql/core@1.10.4/dist/urql-core.js#492)

**Comment by zoontek** [comment]
Yes that's extremely weird. I suppose Metro rewrite code in an odd way.

**Comment by JoviDeCroock** [comment]
I wonder what could be influencing this, is it miss-behaving in inlining the code it removed due to the `replace` of `process.env` 🤔

**Comment by zoontek** [comment]
Wait, I got it! You checked the wrong version (my fault, I shouldn't have posted both)

`f` is declared here: https://www.runpkg.com/?@urql/core@1.11.4/dist/urql-core.js#383 (in `"production" !== process.env.NODE_ENV`) and reused [here](https://www.runpkg.com/?@urql/core@1.11.4/dist/urql-core.js#396). So it's no tied to metro, each bundler using `urql-core.js` will break I think.

**Comment by JoviDeCroock** [comment]
It works in other bundlers the problem seems to be that metro does not account for the code blocks it snips out, they must be doing this as post-processing.

**Comment by zoontek** [comment]
Maybe they just remove the conditional code block without taking care of variables defined inside.

**Comment by zoontek** [comment]
@kitten It works! `hoist_vars` did the trick 🎉

**Comment by kitten** [comment]
This should be released in `@urql/core@1.11.5`, so a quick `yarn upgrade @urql/core` should fix the issue :+1: https://github.com/FormidableLabs/urql/releases/tag/%40urql%2Fcore%401.11.5

**Comment by zoontek** [comment]
Thanks, this was fast!
